contrib/vagrant/provision-minimal.sh

```diff
@@ -1,20 +1,64 @@
 #!/bin/bash
-set -euo pipefail
-IFS=$'\n\t'
 
-sed -i s/^Defaults.*requiretty/\#Defaults\ requiretty/g /etc/sudoers
+function setup {
+	set -euo pipefail
+	IFS=$'\n\t'
 
-# patch incompatible with fail-over DNS setup
-SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
-if [[ -f "${SCRIPT}" ]]; then
-    echo "Removing ${SCRIPT}..."
-    rm "${SCRIPT}"
-    sed -i -e '/^options.*$/d' /etc/resolv.conf
-fi
-unset SCRIPT
+	sed -i s/^Defaults.*requiretty/\#Defaults\ requiretty/g /etc/sudoers
 
-if [ -f /usr/bin/generate_openshift_service ]
-then
-  sudo /usr/bin/generate_openshift_service
-fi
+	# patch incompatible with fail-over DNS setup
+	SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
+	if [[ -f "${SCRIPT}" ]]; then
+	    echo "Removing ${SCRIPT}..."
+	    rm "${SCRIPT}"
+	    sed -i -e '/^options.*$/d' /etc/resolv.conf
+	fi
+	unset SCRIPT
 
+	if [ -f /usr/bin/generate_openshift_service ]
+	then
+	  sudo /usr/bin/generate_openshift_service
+	fi
+}
+
+function installOpenShift {
```

Comment:
This is probably too opinionated.  The registry isn't required for example.  Today the use cases start at
vagrant up
vagrant ssh -c 'sudo systemctl start openshift'

And they have a running ui/system.

You also can't do this for every type.  This would only be applicable to an INST image but not an OS or DEPS image.

We could (and have) made other VMs that bake this level of config into the instances.  But they have all been abandoned in favor of the CDK at this point.